src/csvReaderWrapper.js

```diff
@@ -19,23 +19,35 @@ export const parseCSV = (csvfile, download, useDefaultColumnDataType) => {
   });
 };
 
-const isCellValid = (cell) => {
-  return cell !== undefined && cell !== "";
-}
+const cleanData = (data) => {
+  var cleanedData = []
 
-const isRowValid = (row) => {
-  var cells = Object.values(row);
-  return cells.every(isCellValid)
-}
+  for (var row of data) {
+    var cleanedRow = tryGetCleanedRow(row);
+    if (cleanedRow !== null) {
+      cleanedData.push(cleanedRow);
+    }
+  }
 
-const cleanData = (data) => {
-  var cleanedData = data.filter(row => isRowValid(row));
   return cleanedData;
 }
 
-const countRemovedRows = (originalData, cleanedData) => {
-  var removedRowsCount = originalData.length - cleanedData.length;
-  store.dispatch(setRemovedRowsCount(removedRowsCount));
+const tryGetCleanedRow = (row) => {
+  for (var cellName in row) {
+    if (!isCellValid(row[cellName])) {
```

Comment:
nit:`row[cellName]` is reused so can be extracted to a variable

Reply:
Sounds good!